augur/utils.py

```diff
@@ -809,3 +809,84 @@ def is_augur_version_compatable(version):
     current_version = packaging_version.parse(get_augur_version())
     this_version = packaging_version.parse(version)
     return this_version.release[0] == current_version.release[0]
+
+def read_bed_file(bed_file):
+    """Read a BED file and return a list of excluded sites.
+
+    Note: This function assumes the given file is a BED file. On parsing
+    failures, it will attempt to skip the first line and retry, but no
+    other error checking is attempted. Incorrectly formatted files will
+    raise errors.
+
+    Parameters
+    ----------
+    bed_file : str
+        Path to the BED file
+
+    Returns:
+    --------
+    list[int]:
+        Sorted list of unique zero-indexed sites
+    """
+    mask_sites = []
+    try:
+        bed = pd.read_csv(bed_file, sep='\t', header=None, usecols=[1,2],
+                          dtype={1:int,2:int})
+    except ValueError:
+        # Check if we have a header row. Otherwise, just fail.
+        bed = pd.read_csv(bed_file, sep='\t', header=None, usecols=[1,2],
+                          dtype={1:int,2:int}, skiprows=1)
+        print("Skipped row 1 of %s, assuming it is a header." % bed_file)
+    for _, row in bed.iterrows():
+        mask_sites.extend(range(row[1], row[2]))
+    return sorted(set(mask_sites))
```

Comment:
This approach of only allowing parse errors on the header is really nice.

Reply:
Yeah, I think it's closer to the Principle of Least Surprise re: masking sites - just ignoring random errors in the bed file seems like it could hide some ugly surprises.